btcutil/psbt/taproot.go

```diff
@@ -0,0 +1,219 @@
+package psbt
+
+import (
+	"bytes"
+
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+)
+
+const (
+	// schnorrSigMinLength is the minimum length of a Schnorr signature
+	// which is 64 bytes.
+	schnorrSigMinLength = schnorr.SignatureSize
+
+	// schnorrSigMaxLength is the maximum length of a Schnorr signature
+	// which is 64 bytes plus one byte for the appended sighash flag.
+	schnorrSigMaxLength = schnorrSigMinLength + 1
+)
+
+// TaprootScriptSpendSig encapsulates an individual Schnorr signature for a
+// given public key and leaf hash.
+type TaprootScriptSpendSig struct {
+	XOnlyPubKey []byte
+	LeafHash    []byte
+	Signature   []byte
+	SigHash     txscript.SigHashType
+}
+
+// checkValid checks that both the pubkey and the signature are valid.
+func (s *TaprootScriptSpendSig) checkValid() bool {
+	return validateXOnlyPubkey(s.XOnlyPubKey) &&
+		validateSchnorrSignature(s.Signature)
+}
+
+// EqualKey returns true if this script spend signature's key data is the same
+// as the given script spend signature.
+func (s *TaprootScriptSpendSig) EqualKey(other *TaprootScriptSpendSig) bool {
+	return bytes.Equal(s.XOnlyPubKey, other.XOnlyPubKey) &&
```

Comment:
What about the sighash information?

Reply:
The key only consists of the x-only-pubkey and the leaf hash. The signature and sighash is in the value.